cookbooks/providers/application/providers/default.rb

```diff
@@ -43,6 +43,18 @@
     end
   end
 
+  log %{Enable logrotation...}
+  template "/etc/logrotate.d/nginx-#{new_resource.name}" do
+    cookbook 'application'
+    owner "root"
+    group "root"
+    source "nginx/logrotate.erb"
```

Comment:
I know there is an option where you can use the template of another cookbook. I think it's called `cookbook "passenger"`. So you can keep it DRY.